io/src/main/scala/fs2/io/io.scala

```diff
@@ -74,7 +74,7 @@ package object io {
     F.blocking(is.read(buf)).map { numBytes =>
       if (numBytes < 0) None
       else if (numBytes == 0) Some(Chunk.empty)
-      else if (numBytes < buf.size) Some(Chunk.array(buf.slice(0, numBytes)))
```

Comment:
I think this may have been leftover from when the `buf` was shared across multiple calls, making a copy necessary.

Reply:
But is the change correct?

Reply:
Yep, b/c the only use of this private method is one which allocates a fresh buffer on each invocation